src/liballoc/rc.rs

```diff
@@ -573,10 +573,11 @@ impl<T: ?Sized> Rc<T> {
     /// Constructs an `Rc` from a raw pointer.
     ///
     /// The raw pointer must have been previously returned by a call to a
-    /// [`Rc::into_raw`][into_raw].
+    /// [`Rc::into_raw`][into_raw] using the same `T`.
     ///
-    /// This function is unsafe because improper use may lead to memory problems. For example, a
-    /// double-free may occur if the function is called twice on the same raw pointer.
+    /// This function is unsafe because improper use may lead to memory unsafety,
+    /// even if `T` is never accessed. For example, a double-free may occur if the function is
```

Comment:
I guess this means "if `Rc<T>` is never accessed"`?

Reply:
True as well, although the pitfall here is to assume accessing `Rc<T>` itself was safe if `T` is never touched.

> even if `Rc<T>` or `T` is never accessed.

?

Reply:
This whole paragraph seems a bit strange to me. Underspecified in a way. So what are the conditions for this method to be safe? 

- What's specified in the first paragraph (from a previous `into_raw` call)
- Only call `from_raw` once per raw pointer.
- `T` must (still) be initialized so that it can be dropped.

That's it, right? Shouldn't we explicitly document that?  We can still mention the double free later.

Reply:
It's worth noting (in my opinion) that the `from_raw` once per raw pointer makes relatively little sense, and IMO represents more hassle than it's worth. It means you can't technically store a pointer and then clone off copies of it, i.e., something like this would not be valid, but I think should be.

Other than that, I don't see any issues with the other points.

```rust
unsafe fn clone_arc(ptr: *const T) -> Arc<T> {
    let a = Arc::from_raw(ptr);
    let copy = a.clone();
    mem::forget(a);
    copy
}

let ptr = Arc::into_raw(some_arc);
clone_arc(ptr);
clone_arc(ptr);
clone_arc(ptr);
```

Reply:
Good point. I guess we can just say "the user of `from_raw` has to make sure a specific `T` is only dropped once". Then it's the users responsibility.